controller/instance_manager_controller.go

```diff
@@ -468,26 +468,61 @@ func (imc *InstanceManagerController) syncInstanceManagerPDB(im *longhorn.Instan
 		return err
 	}
 
+	clusterAutoscalerEnabled, err := imc.ds.GetSettingAsBool(types.SettingNameKubernetesClusterAutoscalerEnabled)
+	if err != nil {
+		return err
+	}
+
 	// When current node is unschedulable, it is a signal that the node is being cordoned/drained.
 	if unschedulable {
 		if imPDB == nil {
 			return nil
 		}
-		canDelete, err := imc.canDeleteInstanceManagerPDB(im)
+
+		imc.logger.Debugf("Node %v is marked unschedulable, try to remove %v PDB", imc.controllerID, im.Name)
+		canDelete, availableNode, err := imc.canDeleteInstanceManagerPDB(im)
 		if err != nil {
 			return err
 		}
 		if !canDelete {
+			if clusterAutoscalerEnabled && im.Spec.Type == longhorn.InstanceManagerTypeReplica {
+				if availableNode == "" {
+					return errors.Errorf("cannot find healthy replica on another node")
+				}
```

Comment:
Should we error out here?

Reply:
Here we are expecting this node would have the only replica IM PDB. When a node is marked unschedulable we want another available node with the volume replica to re-create the replica IM PDB.

Reply:
Yes, we should block/avoid deleting the PDB for this IM that contains the only replica of a volume. But it does not mean that we need to fail the **whole IM controller reconcile loop**.

Reply:
I see. Fixed.